Merge master into stable V1.1.1

glynhudson opened the pull request:
I'm not sure if this is related but a user reported that switching to master branch solved his issues with the dashboard module. Could current master be merged into stable? 

https://community.openenergymonitor.org/t/dashboard-missing-in-local-emoncms/4961/3

I'm not sure why there is a merge conflict? This should not happen since stable should always be lagging master. Could you take a look and merge is possible @Paul-Reed

Paul-Reed commented:
@glynhudson @TrystanLea - Spent over an hour on this tonight, and couldn't resolve the merge conflict.
Eventually, I deleted the `stable` branch and recreated it, so now both branches are shown as equal and looks good in github...
However, retrospectively I'm concerned that this was the right thing to do, as existing users will be tracking the original `stable` branch. What do you think?

Paul

glynhudson commented:
Hi @Paul-Reed 

Deleting the stable branch has caused a problem since all emonPi's are now running the stable branch and since it was deleted then re-created it won't be possible for them to update since the fast-forward links have been severed.  With the help of @TrystanLea we reverted stable back to the point before it was deleted, fixed the merge conflicts then pushed it back. We've just tested and now emonPi's can update again. 

Going forward please ensure nothing is committed to stable without it going into mater fist so no merge conflicts occur. I've also made Stable a protected branch will will not allow it to be deleted or force pushed. 

To merge into stable please create a pull request from master and create a release log. 

Thanks again for your help reviewing PR's for the dashboard module, it's fantastic how many contributions the dashboard modules gets :+1:

Paul-Reed commented:
Hi Glyn as far as I'm aware, I've not committed anything to the stable branch, and every PR has been merged into master.
Can you identify what was committed to the stable branch to create this problem?
I spent considerable time double checking every commit that I had made, but all of them confirmed to me that they were made to the master branch.

It would be good to know, because if I have made an error, I want to be sure that I learn from it, and don't repeat it!

Paul

glynhudson commented:
I think it was this commit: https://github.com/emoncms/dashboard/commit/4357466db4f3f24b909f52ab3c6902ad39fdfa23

I would recomend always making a PR using github.com to merge master into stable this allows us to easily see what's about to be merged before it's actually merged and create an audit trail e.g. got to:

https://github.com/emoncms/dashboard/compare/master...stable

Then click create PR, give the PR the name of the release version number, obviously there are no changes to compare at the moment!

See emoncms core stable branch PR merges as example: https://github.com/emoncms/emoncms/pull/707

Paul-Reed commented:
I'm not sure what went wrong with https://github.com/emoncms/dashboard/commit/4357466db4f3f24b909f52ab3c6902ad39fdfa23 as I did use github.com, created a PR but merged it pretty much straight away (as IE11 browser's were crashing), but obviously something went wrong.
It's probably better for me to leave branch merging to you in future, or at least leave the PR for you to check & merge.
Apologies to you & Trystan for messing this up.

Paul

glynhudson commented:
No worries, we've all done it!

On 7 September 2017 at 19:12, Paul <notifications@github.com> wrote:

> I'm not sure what went wrong with 4357466
> <https://github.com/emoncms/dashboard/commit/4357466db4f3f24b909f52ab3c6902ad39fdfa23>
> as I did use github.com, created a PR but merged it pretty much straight
> away (as IE11 browser's were crashing), but obviously something went wrong.
> It's probably better for me to leave branch merging to you in future, or
> at least leave the PR for you to check & merge.
> Apologies to you & Trystan for messing this up.
>
> Paul
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/emoncms/dashboard/pull/135#issuecomment-327878815>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAuUPA8G3Hb-1P-1aRJiSlIJ2FfHzJTyks5sgDChgaJpZM4PL_8_>
> .
>



-- 
Glyn Hudson

http://megni.co.uk
http://adventuresplusnorthwales.blogspot.com
